src/components/ModalForm.js

```diff
@@ -19,101 +18,127 @@ export default function ModalForm(props) {
   const [creator, updateCreator] = useState('');
   const [minTime, updateMinTime] = useState('');
   const [maxTime, updateMaxTime] = useState('');
+  const [validated, setValidated] = useState(false);
 
   // Used by the submit button to filter the recipes displayed and close modal
-  const onSubmit = () => {
-    onPageChange(1);
-    setName(recipeName);
-    setCreator(creator);
-    setMinTime(minTime);
-    setMaxTime(maxTime);
-    setModal(false);
-    setIngredients(ingredients);
+  const onSubmit = (event) => {
+    const form = event.currentTarget;
+    if (!form.checkValidity()) {
+      event.preventDefault();
+      event.stopPropagation();
+    } else {
+      onPageChange(1);
+      setName(recipeName);
+      setCreator(creator);
+      setMinTime(minTime);
+      setMaxTime(maxTime);
+      setModal(false);
+      setIngredients(ingredients);
+    }
+    setValidated(true);
   };
 
   return (
     <>
       <Modal
         show={modalOpen}
         onHide={() => setModal(false)}
-        centered="true"
+        centered
```

Comment:
note on whole PR - please remember to have your PR/commit messages be in present case. You also want the more important part of the message to be in front. Here, the majority of complex code is on form validation so you want to put that first so your reviewer's focus can be on that. Also, if there are other people scanning through the changes for when a particular bug has been introduced, they can scan the commit messages quicker.

E.g. "Implement validation for ingredients form. Also update style"